src/main/java/com/epam/ta/reportportal/dao/TestItemRepository.java

```diff
@@ -185,4 +185,25 @@
 	boolean hasStatusNotEqualsWithoutStepItem(@Param("parentId") Long parentId, @Param("stepId") Long stepId,
 			@Param("status") StatusEnum status);
 
+	/**
+	 * Finds root(without any parent) {@link TestItem} with specified {@code name} and {@code launchId}
+	 *
+	 * @param name     Name of {@link TestItem}
+	 * @param launchId ID of {@link Launch}
+	 * @return {@link Optional<TestItem>} if it exists, {@link Optional#empty()} if not
+	 */
+	@Query("select t from TestItem t where t.name=:name and t.parent is null and t.launch.id=:launchId")
```

Comment:
make it with upper case

Reply:
check if it can be written with native query without joins

Reply:
fixed